services/spar/src/Spar/Scim/User.hs

```diff
@@ -530,7 +546,22 @@ createValidScimUser tokeninfo@ScimTokenInfo {stiTeam} vsu@(ST.ValidScimUser veid
         let new = ST.scimActiveFlagToAccountStatus old (Scim.unScimBool <$> active)
             active = Scim.active . Scim.value . Scim.thing $ storedUser
         when (new /= old) $ BrigAccess.setStatus buid new
+
+      lift $ ScimExternalIdStore.insertStatus stiTeam veid buid ScimUserCreated
       pure storedUser
+  where
+    incompleteUserCreationCleanUp :: UserId -> Scim.ScimError -> Scim.ScimHandler (Sem r) ()
+    incompleteUserCreationCleanUp buid e = do
+      -- something went wrong while storing the user in brig
+      -- we can try clean up now, but if brig is down, we can't do much
+      -- maybe retrying the user creation in brig is also an option?
+      -- after clean up we rethrow the error so the handler returns the correct failure
```

Comment:
what about *not* rethrowing, but looking up the status again once or twice?  that'd make for swifter cleanup, especially when the batch job only runs once a day.

Reply:
It only rethrows if `deleteScimUser` fails. AFAICT the deletion is synchronous.